### PR TITLE
Fix logging of drum kit selection

### PR DESCRIPTION
**Pull request by oleg-derevenetz**
Unlike regular musical instruments, where each key of the virtual synthesizer corresponds to a musical note, in drum kits, each key of the virtual synthesizer corresponds to a separate drum sound. Thus, the Program Change event in the case of drums switches not the instrument (an individual drum) itself, but the whole drum kit.

### CODE REVIEW

File: src/engine/xmi2mid.cpp [hunk 1]
```diff
@@ -113,15 +114,7 @@ namespace
...
-            "F#2 Closed Hi Hat",     "G2 High Floor Tom", "G#2 Pedal Hi-Hat",  "A2 Low Tom",        "A#2 Open Hi-Hat", "B2 Low-Mid Tom",    "C3 Hi Mid Tom",
-            "C#3 Crash Cymbal 1",    "D3 High Tom",       "D#3 Ride Cymbal 1", "E3 Chinese Cymbal", "F3 Ride Bell",    "F#3 Tambourine",    "G3 Splash Cymbal",
-            "G#3 Cowbell",           "A3 Crash Cymbal 2", "A#3 Vibraslap",     "B3 Ride Cymbal 2",  "C4 Hi Bongo",     "C#4 Low Bongo",     "D4 Mute Hi Conga",
-            "D#4 Open Hi Conga",     "E4 Low Conga",      "F4 High Timbale",   "F#4 Low Timbale",   "G4 High Agogo",   "G#4 Low Agogo",     "A4 Cabasa",
-            "A#4 Maracas",           "B4 Short Whistle",  "C5 Long Whistle",   "C#5 Short Guiro",   "D5 Long Guiro",   "D#5 Claves",        "E5 Hi Wood Block",
-            "F5 Low Wood Block",     "F#5 Mute Cuica",    "G5 Open Cuica",     "G#5 Mute Triangle", "A5 Open Triangle" };
+    const std::map<uint32_t, const char *> drumKitDescription = { { 0, "Standard" }, { 48, "Orchestral" }, { 49, "Fix Room" } };
```
Comment (ihhub) [inline src/engine/xmi2mid.cpp:117]: Could you please provide a link for this information? What if we use standard MIDI player which includes these sounds?

Comment (oleg-derevenetz) [reply src/engine/xmi2mid.cpp:117]: @ihhub to be honest, I don't know such a link. Only drum set 0 is the "standard" one, others may or may not present (even many of Yamaha hardware synthesizers have just the drum set 0). I found just one soundfont with set 49 so far, and no soundfonts with other two sets requested by some MIDI tracks from the game, so I even don't know their names :)

Comment (oleg-derevenetz) [reply src/engine/xmi2mid.cpp:117]: @ihhub I found only this so far (see the "Other Drum Kits" section):

https://www.pgmusic.com/tutorial_gm.htm

But the described set is very limited, and none of drum kits used by the game are listed (except the Standard one, of course).

**Comment by ihhub** [comment]
@oleg-derevenetz , I am still reviewing it. Please do not think that I am ignoring this pull request :)

**Comment by oleg-derevenetz** [comment]
@ihhub I'm not in a hurry :)

File: src/engine/xmi2mid.cpp (hunk 1 above)
Comment (ihhub) [reply src/engine/xmi2mid.cpp:118]: Shall we then populate this map with other kits according to wiki page?
<html>
<body>


1 | Standard Kit | The only kit specified by General MIDI Level 1
-- | -- | --
9 | Room Kit | Drums recorded with room ambience
17 | Power Kit | More powerful kick and snare sounds
25 | Electronic Kit | Sounds of various electronic drums
26 | TR-808 Kit | Analog drum kit similar to Roland TR-808
33 | Jazz Kit | Softer kick and snare sounds than the Standard Kit
41 | Brush Kit | Many brush sounds added
49 | Orchestral Kit | A collection of concert drums and timpani
57 | Sound FX Kit | A collection of sound effects


</body>
</html>

**Pull request merged**